src/Bridge/Symfony/DependencyInjection/DbToolsExtension.php

```diff
@@ -88,7 +88,7 @@ public function load(array $configs, ContainerBuilder $container): void
         $strategyServices = [];
         foreach ($config['connections'] as $connectionName => $connectionConfig) {
             // Default is handled directly by the storage service.
-            $strategyId = $connectionConfig['storage_filename_strategy'] ?? null;
+            $strategyId = $connectionConfig['storage_filename_strategy'] ?? $config['storage_filename_strategy'] ?? null;
```

Comment:
This is not good, we have a configuration object for each user defined configuration, which falls back on a default configuration object at runtime when fetching its configuration. You should have checked that the default configuration is rightly populated.